GVFS/GVFS.Common/Prefetch/Jobs/ReadFilesJob.cs

```diff
@@ -44,11 +44,24 @@ protected override void DoWork()
                     foreach (string path in this.blobIdToPaths[blobId])
                     {
                         bool succeeded = false;
-                        using (SafeFileHandle handle = NativeFileReader.Open(path))
+
+                        // TODO(Mac): Replace this with a native method as opposed to a slow .NET method
+                        if (!GVFSPlatform.Instance.IsUnderConstruction)
+                        {
+                            using (SafeFileHandle handle = NativeFileReader.Open(path))
+                            {
+                                if (!handle.IsInvalid)
+                                {
+                                    succeeded = NativeFileReader.ReadOneByte(handle, buffer);
+                                }
+                            }
+                        }
+                        else
                         {
-                            if (!handle.IsInvalid)
+                            using (FileStream filestream = new FileStream(path, FileMode.Open, FileAccess.Read))
                             {
-                                succeeded = NativeFileReader.ReadOneByte(handle, buffer);
+                                filestream.ReadByte();
+                                succeeded = true;

```

Comment:
you are assuming ReadByte always succeeds.
Something like - ```succeeded = -1 != filestream.ReadByte();``` would be more correct.

Reply:
-1 is not an error here, it just indicates end of file. I believe you would get that if this is a 0-byte file, and it would still be a success case.